src/evergreen/task.py

```diff
@@ -34,18 +36,35 @@ class Artifact(_BaseEvergreenObject):
     url = evg_attrib("url")
     visibility = evg_attrib("visibility")
     ignore_for_fetch = evg_attrib("ignore_for_fetch")
+    content_type = evg_attrib("content_type")
 
     def __init__(self, json: Dict[str, Any], api: "EvergreenApi") -> None:
         """Create an instance of an evergreen task artifact."""
         super(Artifact, self).__init__(json, api)
 
-    def stream(self) -> Iterable[str]:
+    def stream(self, decode_unicode: bool = True, chunk_size: int = None,) -> Iterable[str]:
         """
         Retrieve an iterator of the streamed contents of this artifact.
 
+        :param decode_unicode: determines if we decode as unicode
+        :param chunk_size: the size of the chunks to be read
         :return: Iterable to stream contents of artifact.
         """
-        return self._api._stream_api(self.url)
+        return self._api._stream_api(
+            self.url,
+            decode_unicode=decode_unicode,
+            chunk_size=chunk_size,
+            is_binary=self._is_binary(),
+        )
+
+    def _is_binary(self) -> bool:
+        """Determine if an artifact is binary based on content_type."""
+        _type, subtype = self.content_type.split("/")
+
+        if _type in _BINARY_TYPES:
+            return True
+        else:
+            return False
```

Comment:
Can we make this assumption? I'm not familiar with this portion of the Evergreen api, is this how Evergreen stores binary data always?

Reply:
evergreen uses this standard for their content_type field of artifacts:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types 
https://www.iana.org/assignments/media-types/media-types.xhtml   
<img width="1016" alt="image" src="https://github.com/evergreen-ci/evergreen.py/assets/57504606/16df29ec-51df-43a2-912f-3d68a8804f99">


i can also check in with the evergreen team about this issue!

Reply:
Gotcha. I think that's fine then. However, I think it would also be useful to allow people to override the behind the scenes binary check if needed. So, it would be great if we added a `is_binary` param to the `stream` function that would override the `_is_binary` function only if specified.